examples/federated-learning/tff/nvflare/README.md

```diff
@@ -0,0 +1,255 @@
+
+![alt_text](../../../../assets/nvflare.svg "Infrastructure overview")
+
+As shown in the preceding diagram, the blueprint helps you to create and configure the following infrastructure components:
+
+- A persistent volume to store the Nvflare workspace
+- Two pods that are the clients that will be connected to the server in the nvidia-client1 and nvidia-client2 namespaces respectively
+- One pod that is the the server that will aggregate all the results from the computation in the nvflare-infra namespace
+
+## Deploy the blueprint
+
+This example builds on top of the infrastructure that the
+[blueprint provides](../../../../README.md), and follows the best practices the
+blueprint establishes.
+
+To deploy the Nvflare demo described in this document, you need to deploy the [Federated learning blueprint](../../../../README.md#deploy-the-blueprint) first. Then, you can deploy the Nvflare demo described in this document.
+
+### 1. Create custom image
+
+For this demo to work, you have to create a custom image with TensorFlow and Nvflare installed. From the main `terraform` folder:
+
+1. Build the example container image locally on your host:
+
+   ```bash
+   NVFLARE_EXAMPLE_CONTAINER_IMAGE_REPOSITORY_HOSTNAME="$(terraform output -raw container_image_repository_fully_qualified_hostname)"
+   export NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID="${NVFLARE_EXAMPLE_CONTAINER_IMAGE_REPOSITORY_HOSTNAME}/$(terraform output -raw container_image_repository_name)/nvflare-tensorflow"
+   export NVFLARE_EXAMPLE_CONTAINER_IMAGE_TAG="0.0.1"
+   NVFLARE_EXAMPLE_CONTAINER_IMAGE_BUILD_CONTEXT_PATH="../examples/federated-learning/tff/nvflare/container-image"
+   NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID_WITH_TAG=${NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID}:${NVFLARE_EXAMPLE_CONTAINER_IMAGE_TAG}
+
+   docker build \
+     --file "${NVFLARE_EXAMPLE_CONTAINER_IMAGE_BUILD_CONTEXT_PATH}/Dockerfile" \
+     --tag "${NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID_WITH_TAG}" \
+     ${NVFLARE_EXAMPLE_CONTAINER_IMAGE_BUILD_CONTEXT_PATH}
+   ```
+
+1. Authenticate Docker with the Artifact Registry repository:
+
+   ```bash
+   gcloud auth configure-docker \
+       "${NVFLARE_EXAMPLE_CONTAINER_IMAGE_REPOSITORY_HOSTNAME}"
+   ```
+
+1. Push the container image to the Artifact Registry repository:
+
+   ```bash
+   docker image push "${NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID_WITH_TAG}"
+   ```
+
+Once the image has beed built, modify the `kustomization.yaml.tpl` file to add the new name and new tag of the image:
+
+```bash
+envsubst < ../examples/federated-learning/tff/nvflare/templates/kustomization.yaml.tpl > ../examples/federated-learning/tff/nvflare/kustomization.yaml
+```
+
+### 2. Create infrastructure to store the models
+
+All the models generated will be stored in a Cloud storage bucket mounted by each pod. To create the Cloud storage and give the right permissions, from the `terraform` folder, add the flag `nvflare = true` to your tfvars file and run :
```

Comment:
This assumes that the reader is using a `tfvars` file, which might not be the case. Refactor the last part of this sentence as something like: `Initialize the following Terraform variables`

Example: https://github.com/GoogleCloudPlatform/federated-learning/blob/main/examples/federated-learning/tff/distributed-fl-simulation-k8s/README.md#containers-running-in-different-namespaces-in-the-same-gke-cluster